apps/src/lib/levelbuilder/lesson-editor/FindProgrammingExpressionDialog.jsx

```diff
@@ -0,0 +1,217 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+
+import Button from '@cdo/apps/templates/Button';
+import DialogFooter from '@cdo/apps/templates/teacherDashboard/DialogFooter';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+
+import LessonEditorDialog from './LessonEditorDialog';
+
+const SearchForm = function(props) {
+  return (
+    <form className="form-search">
+      <input
+        type="text"
+        className="input-large search-query"
+        onChange={props.onSearch}
+        placeholder="search by name"
+      />
+      <select
+        className="input-small search-query"
+        style={{marginLeft: 4}}
+        onChange={props.onFilter}
+      >
+        <option value="">filter</option>
+        {/* TODO: these should probably be sourced from somewhere, rather than hardcoded */}
+        <option value="applab">Applab</option>
+        <option value="gamelab">Gamelab</option>
+        <option value="spritelab">Spritelab</option>
+        <option value="weblab">Weblab</option>
+      </select>
+    </form>
+  );
+};
+
+SearchForm.propTypes = {
+  onSearch: PropTypes.func.isRequired,
+  onFilter: PropTypes.func.isRequired
+};
+
+const ProgrammingExpressionTable = function(props) {
+  if (
+    !props.programmingExpressions ||
+    props.programmingExpressions.length === 0
+  ) {
+    return null;
+  }
+
+  // TODO what should we actually do here?
+  const defaultColor = '#0094ca';
+
+  // TODO implement pagination
+  // As a temporary crutch until we have time to get pagination working, here's what we do:
+  // Display a maximum of 10 results.
+  // If we are given more than 10 results, inform the user that there
+  // are more results and suggest that they narrow the search term.
+  const tooManyResults = props.programmingExpressions.length > 10;
+  const displayExpressions = props.programmingExpressions.slice(0, 10);
+
+  return (
+    <table className="table table-striped table-bordered">
+      <thead>
+        <tr>
+          <th />
+          <th>Expression</th>
+          <th>Environment</th>
+        </tr>
+      </thead>
+      <tbody>
+        {displayExpressions.map(programmingExpression => (
+          <tr key={programmingExpression.uniqueKey}>
+            <td>
+              <input
+                type="radio"
+                name="programmingExpression"
+                value={programmingExpression.uniqueKey}
+                onChange={props.handleSelect}
+              />
+            </td>
+            <td>
+              <SafeMarkdown
```

Comment:
We should probably make a component to wrap this since we will need to use it in a couple different places to display expressions like this. I can do this as follow up work.

Reply:
I'm actually building something for this as part of the "better default color behavior" fix in this PR

Reply:
Created in this PR: https://github.com/code-dot-org/code-dot-org/pull/39716.  I also fixed the default color issue you were talking about above I think